### PR TITLE
fix: mypy "has-type" check

### PR DESCRIPTION
**Pull request by bartsanchez**

### CODE REVIEW

File: src/arch/z80/peephole/evaluator.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ class Number:
 
     def __init__(self, value):
         if isinstance(value, Number):
-            self.value = value.value
+            self.value: Any = value.value
```
Comment (boriel) [inline src/arch/z80/peephole/evaluator.py:109]: Should be:
```python
self.value: int | None = value.value
```

File: src/arch/z80/optimizer/basicblock.py [hunk 2]
```diff
@@ -440,7 +440,7 @@ def optimize(self, patterns_list):
             "c": str(self.cpu.C) if self.cpu.C is not None else new_tmp_val(),
             "z": str(self.cpu.Z) if self.cpu.Z is not None else new_tmp_val(),
         }.get(x.lower(), new_tmp_val())
-        evaluator.UNARY[FN.IS_REQUIRED] = lambda x: self.is_used([x], i + len(p.patt))
+        evaluator.UNARY[FN.IS_REQUIRED] = lambda x: self.is_used([x], i + len(p.patt))  # type: ignore[has-type]
```
Comment (boriel) [inline src/arch/z80/optimizer/basicblock.py:443]: Here `i` is `int` and `p` is of type `OptPattern`.

Suggestion:
 * add `from src.arch.z80.peephole.engine import OptPattern` at the beginning of the file.
 * change the function signature to: `def optimize(self, patterns_list: Sequence[OptPattern]) -> None:`
  * Add these to lines below:
   `i: int`
   `p: OptPattern`
   
 This way we can get rid of the `# type: ignore` line :wink:

**Review by boriel — CHANGES_REQUESTED** [review]
I think it's better to type the missing variables.
What do you think?

File: src/arch/z80/peephole/evaluator.py (hunk 1 above)
Comment (bartsanchez) [reply src/arch/z80/peephole/evaluator.py]: Great, changed!

File: src/arch/z80/optimizer/basicblock.py (hunk 2 above)
Comment (bartsanchez) [reply src/arch/z80/optimizer/basicblock.py:447]: Added the variables below the function header.

**Review by boriel — APPROVED** [review]

**Pull request merged**